src/main/java/com/twilio/oai/api/JavaApiResourceBuilder.java

```diff
@@ -147,6 +156,14 @@ public ApiResourceBuilder updateOperations(Resolver<CodegenParameter> codegenPar
         return this;
     }
 
+    private CodegenParameter copySecuritySchemaToSubLevels(CodegenParameter item, CodegenOperation co) {
+        String schema = (String) co.vendorExtensions.get("x-http-class-prefix");
+        if(schema != null){
+            item.vendorExtensions.put("x-http-class-prefix", schema);
```

Comment:
Why we need to set this for each parameter ?
Is this done for Request prefix ? Lets not add this even we have change our Request design.
Please note: if you can improve how requests are defined or renamed, rename it.

Reply:
{{#headerParams.0}}
    private void addHeaderParams(final {{vendorExtensions.x-http-class-prefix}}Request request) {
{{>headerParams}}

For the above code snippet, we need to make the parameter Request choosen based on authentication method. Since we say headerParams.0 this is the first header parameter. So the variables set for the first parameter only will be considered inside this section. 

One way to resolve this is to set new variables like hasHeaderParams, hasQueryParams etc at operation level and then use it, should we do like that?

Reply:
Tried these things

1. Checked if vendor extensions are not present at current level, will it check at the outer level ? NO
2. Can we call like co.queryParams.0 - No, this is also not working
3. Add new vendor extensions like hasHeaderParams, hasQueryParams etc - yes this is working, implemented